CesiumGltfContent/src/GltfUtilities.cpp

```diff
@@ -901,4 +1228,265 @@ void GltfUtilities::compactBuffer(
+createPositionView(
+    const Model& model,
+    const Accessor& accessor,
+    TCallback&& callback) {
+  assert(accessor.type == Accessor::Type::VEC3);
+
+  switch (accessor.componentType) {
+  case Accessor::ComponentType::BYTE:
+    return callback(AccessorView<AccessorTypes::VEC3<int8_t>>(model, accessor));
+  case Accessor::ComponentType::UNSIGNED_BYTE:
+    return callback(
+        AccessorView<AccessorTypes::VEC3<uint8_t>>(model, accessor));
+  case Accessor::ComponentType::SHORT:
+    return callback(
+        AccessorView<AccessorTypes::VEC3<int16_t>>(model, accessor));
+  case Accessor::ComponentType::UNSIGNED_SHORT:
+    return callback(
+        AccessorView<AccessorTypes::VEC3<uint16_t>>(model, accessor));
+  case Accessor::ComponentType::UNSIGNED_INT:
+    return callback(
+        AccessorView<AccessorTypes::VEC3<uint32_t>>(model, accessor));
+  case Accessor::ComponentType::FLOAT:
+    return callback(AccessorView<AccessorTypes::VEC3<float>>(model, accessor));
+  default:
+    return callback(AccessorView<AccessorTypes::VEC3<float>>(
+        AccessorViewStatus::InvalidComponentType));
+  }
+}
+
+std::optional<glm::dvec3> intersectRayScenePrimitive(
+    const CesiumGeometry::Ray& ray,
+    const CesiumGltf::Model& model,
+    const CesiumGltf::MeshPrimitive& primitive,
+    const Accessor& positionAccessor,
+    const glm::dmat4x4& primitiveToWorld,
+    bool cullBackFaces,
+    std::vector<std::string>& warnings) {
+  glm::dmat4x4 worldToPrimitive = glm::inverse(primitiveToWorld);
+  CesiumGeometry::Ray transformedRay = ray.transform(worldToPrimitive);
+
+  // Ignore primitive if ray doesn't intersect bounding box
+  const std::vector<double>& min = positionAccessor.min;
+  const std::vector<double>& max = positionAccessor.max;
+
+  double t;
+  if (!CesiumGeometry::IntersectionTests::rayAABBParametric(
+          transformedRay,
+          CesiumGeometry::AxisAlignedBox(
+              min[0],
+              min[1],
+              min[2],
+              max[0],
+              max[1],
+              max[2]),
+          t)) {
+    return std::optional<glm::dvec3>();
+  }
+
+  double tClosest = -1;
+
+  // Support all variations of position component types
+  //
+  // From the glTF spec...
+  // "Floating-point data MUST use IEEE-754 single precision format."
+  //
+  // Yet, the KHR_mesh_quantization extension can specify more
+  //
+  createPositionView(
+      model,
+      positionAccessor,
+      [&transformedRay,
+       &model,
+       &primitive,
+       cullBackFaces,
+       &tClosest,
+       &warnings](const auto& positionView) {
+        // Bail on invalid view
+        if (positionView.status() != AccessorViewStatus::Valid) {
+          warnings.push_back(
+              "Skipping mesh with an invalid position component type");
+          return;
+        }
+
+        bool hasIndexedTriangles = primitive.indices != -1;
+        if (hasIndexedTriangles) {
+          assert(primitive.indices >= 0);
```

Comment:
An assertion is not appropriate here because this is user-provided data. You must check - even in release builds - that indices is valid before using it to index into model.accessors.

There's a getSafe method on Model to make this easy.

Reply:
Agreed. Looks like this validation slipped through the cracks.

Changes committed.